### ISSUE TITLE
Query Invoice Tests Non Canonical - 07/12/18 18:29 - 5004

### ISSUE THREAD

**Issue by SME-Issues** [SME-Issues/issues#2007]
**Query Invoice Tests Non Canonical**
- Total: 788
- Passed: 617
  - **Pass: 537 (82%)**
  - Not Understood: 116
  - Error (not understood): 16
- Failed but Understood: 119 (18%)

**Comment by SME-Issues** [comment]
### Understood and failed but likely higher priority



|Expressions|Notes|
|---|---|
| _am I waiting on any payments_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Response: _Let's check that out for you…{nl} I've scanned the digits and since 8th November 2018 your revenue was eight payments totalling £11,615.26._</li></ul> |
| _by what date should I pay alpha ltd_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Response: _Right, let me look into that…{nl} I've evaluated the state of affairs and your last payment to Alpha Ltd was £662.84 on the 2nd October 2018. You have three unpaid invoices from them totalling £2,273.50, all of which are overdue._</li></ul> |
| _does anyone expect payment from me today_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find one of entities {bill_from, pay_to} but found only {datetime, text}.</li><li>Response: _OK, I'm on this now…{nl} I've looked into this and there were no outgoing transactions for today._</li></ul> |
| _how much does my landlord want this month_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find one of entities {bill_from, pay_to} but found only {date_mod, datetime, text, pay_any}.</li><li>Response: _I can't find that company in your account._</li></ul> |
| _how much money from me are people waiting on_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Response: _Right, let's see just now…{nl} I can confirm since 8th November 2018 your outgoings were nine payments totalling £4,331.17._</li></ul> |
| _I'd like to see every sale I've made_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'debtor' but found only {paid_yet, all, text}</li><li>Response: _Okeydokes. Let's take a look…{nl} I can see your last payment was to Service Charges for £6.50 on the 12th December 2018._</li></ul> |
| _show the unresolved invoices_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'open_mod' but found only {dt, closed_mod, pay_type, text, date_in, date_mod, datetime}</li><li>Response: _Let's check that out for you…{nl} I've just checked the up-to-the-minute situation and since 8th November 2018 you paid seven invoices totalling £2,974.69._</li></ul> |
| _what kind of money is my landlord demanding this month_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find one of entities {bill_from, pay_to} but found only {text, date_mod, datetime, wp, pay_any}.</li><li>Response: _I can't find that company in your account._</li></ul> |
| _what will be the value of screw small businesses this year_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find one of entities {bill_to, pay_from} but found only {date_mod, datetime, text, pay_any}.</li><li>Response: _I can't find that company in your account._</li></ul> |
| _when is the money from ABC ltd coming in_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'open_mod' but found only {when, text, pay_from, money}</li><li>Response: _Let's have a peek…{nl} I've just queried for you and you have never received payment from ABC Ltd. You have one unpaid bill to them which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _when will I get the money in from ABC ltd_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'open_mod' but found only {when, text, pay_from, money}</li><li>Response: _OK hold on, I'm checking now…{nl} I've looked into this and you have never received payment from ABC Ltd. You have one unpaid bill to them which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _when will I recieve the money from ABC ltd_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'open_mod' but found only {when, text, pay_from, money}</li><li>Response: _Right, let me just look…{nl} All right, I've taken stock and you have never received payment from ABC Ltd. You have one unpaid bill to them which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _which bills do you think we should pay this week_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Response: _OK hold on, I'm checking now…{nl} I've looked into this and this week you paid one utility bill for the amount of £931.56 which was paid on the 5th December 2018._</li></ul> |
| _which invoices should we pay on Wed_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Expected to find entity 'open_mod' but found only {datetime, text, pay_to, paid_yet, dt, pay_type}</li><li>Response: _Let me check that out…{nl} So I've just looked over the finances and there were no outgoing creditor invoice transactions on Wednesday (5th December)._</li></ul> |
| _which supplier invoices have yet to be paid_ | <ul><li>expected intent to be `query_invoices` but found `query_payment`</li><li>Response: _I'm determining that now…{nl} I've looked into this and your last paid an invoice on the 2nd December 2018 for the amount of £159.93._</li></ul> |
| _report any bills that are due this month_ | <ul><li>expected intent to be `query_invoices` but found `set_date`</li><li>Expected to find entity 'open_mod' but found only {date_mod, datetime, text}</li><li>Response: _OK hold on, I'm checking now…{nl} I've performed a quick investigation and in December 2018 your expenditure was seven payments totalling £3,630.68._</li></ul> |
| _do I need to chase anyone up about money they owe_ | <ul><li>Expected a single QueryContext but found two</li><li>Other than 'extras and confirmations' expected a single chunk but found 2</li><li>Response: _OK, let me take a look…{nl} So I've just looked over the finances and the overdue amount you are owed from all clients is four invoices totalling £1,478.27. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _give me a bill that I need to pay by next tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "11th December 2018" with a length of 18, but "Tuesday 11th December" has a length of 21.</li><li>Response: _Righty ho, let me investigate immediately…{nl} I've had a look at your account details and the amount you owe to all suppliers on Tuesday 11th December is seven unpaid bills totalling £3,954.81, all of which are overdue._</li></ul> |
| _give me all the bills I have to pay before next Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "11th December 2018" with a length of 18, but "Tuesday 11th December" has a length of 21.</li><li>Response: _Right, let's see just now…{nl} I've explored and the amount you owe to all suppliers on Tuesday 11th December is seven unpaid bills totalling £3,954.81, all of which are overdue._</li></ul> |
| _pass me one of the bills that I need to pay by next Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "11th December 2018" with a length of 18, but "Tuesday 11th December" has a length of 21.</li><li>Response: _Right, let's see just now…{nl} I've explored and the amount you owe to all suppliers on Tuesday 11th December is seven outstanding bills totalling £3,954.81, all of which are overdue._</li></ul> |
| _have we got much to pay on Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "4th September 2018" with a length of 18, but "Tuesday (4th September)" has a length of 23.</li><li>Response: _Ok, I'm examining that now…{nl} I've inspected and you have no outstanding outgoing invoices on Tuesday (4th September)._</li></ul> |
| _add up how much I'll have to pay on Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "Tuesday 4th September" with a length of 21, but "Tuesday (4th September)" has a length of 23.</li><li>Response: _Right, let's see just now…{nl} I've inspected and you have no unpaid outgoing invoices on Tuesday (4th September)._</li></ul> |
| _am I going to be paying a lot on Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DateRangeOutputText to be "Tuesday 4th September" with a length of 21, but "Tuesday (4th September)" has a length of 23.</li><li>Response: _I'll look right now…{nl} I can confirm you have no unpaid outgoing invoices on Tuesday (4th September)._</li></ul> |
| _show me the unpaid bills from march_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-03-31>, but found <2019-03-31>.</li><li>Response: _Let's have a peek…{nl} I've performed a quick investigation and the amount you owe to all suppliers in March 2019 is nine open bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients in March 2019 is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _do I owe kappa clowns anything for Sept_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-09-30>, but found <2019-09-30>.</li><li>Response: _I'm afraid I didn't quite get all of that, however I can't find that company in your records._</li></ul> |
| _what do I owe kappa clowns for Sept_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-09-30>, but found <2019-09-30>.</li><li>Response: _I can't find that company in your records._</li></ul> |
| _will I have much to pay on Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-12-11>, but found <2018-12-04>.</li><li>Response: _I'll look right now…{nl} I can confirm the amount you owe to all suppliers on Tuesday (4th December) is seven unpaid bills totalling £3,954.81, all of which are overdue._</li></ul> |
| _any invoices with payments due on Wed_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-12-12>, but found <2018-12-05>.</li><li>Expected to find entity 'open_mod' but found only {datetime, text, paid_query, dt, pay_type}</li><li>Response: _I'll find out right away…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers on Wednesday (5th December) is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients on Wednesday (5th December) is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _who needs to pay me on friday_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-12-14>, but found <2018-12-07>.</li><li>Response: _I'll find out right away…{nl} I've just checked the up-to-the-minute situation and the amount you are owed from all clients on Friday (7th December) is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _who owes me on friday_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-12-14>, but found <2018-12-07>.</li><li>Response: _Let's have a peek…{nl} I've inspected and the amount you are owed from all clients on Friday (7th December) is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _What am I going to be paying next month_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2018-12-31>, but found <2019-01-31>.</li><li>Response: _OK hold on, I'm checking now…{nl} I've had a quick look and the amount you owe to all suppliers in January 2019 is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do I need to make any payments on 2 October_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.DueDateQueryEnd to be <2019-10-02>, but found <2018-10-02>.</li><li>Response: _Happy to help…{nl} I can confirm the amount you owe to all suppliers on the 2nd October 2018 is one unpaid bill which is for the amount of £800.84 and was due on the 2nd October 2018._</li></ul> |
| _am I still due anything from eta widges_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "eta widgets" with a length of 11, but "eta ridges" has a length of 10.</li><li>Expected {pay_from} entity with value 'eta widgets' but found only 'eta ridges'</li><li>Response: _Regrettably, I didn't recognise everything you said, however I can't find that company in your records._</li></ul> |
| _fetch the invoices for action ai_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "action ai", but found <null>.</li><li>Expected to find entity 'pay_any' but found only {dt, pay_type, text}</li><li>Response: _OK hold on, I'm checking now…{nl} I've determined how things are. The amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _give me the invoices for action ai_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "action ai", but found <null>.</li><li>Expected to find entity 'pay_any' but found only {dt, pay_type, text}</li><li>Response: _I'll find out right away…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _I'd like to see the invoices for action ai_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "action ai", but found <null>.</li><li>Expected to find entity 'pay_any' but found only {dt, pay_type, text}</li><li>Response: _OK, let's see…{nl} So I've just looked over the finances and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _is CYGB going to owe anything_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "cybg", but found <null>.</li><li>Expected to find one of entities {bill_to, pay_from} but found only {unknown_entity, text, paid_query}.</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _how is santander gonna owe me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "santander", but found <null>.</li><li>Entity 'pay_from' does not contain any value but expected {santander}</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _how much is santander going to owe me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.EntityName to be equivalent to "santander", but found <null>.</li><li>Entity 'pay_from' does not contain any value but expected {santander}</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you are owed from all customers is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _any invoices due need paying this Wed_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.HasDateCriteria to be true, but found False.</li><li>Expected to find entity 'datetime' but found only {open_mod, text, paid_query, dt, pay_type}</li><li>Response: _Ok, I'm examining that now…{nl} I've determined how things are. The amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _do I have a lot of money due in august_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.HasDateCriteria to be true, but found False.</li><li>Expected to find one of entities {bill_to, pay_from} but found only {text, open_mod}.</li><li>Response: _Right, let's see just now…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _does it  look like I'm going to have a lot of money due in august_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.HasDateCriteria to be true, but found False.</li><li>Expected to find one of entities {bill_to, pay_from} but found only {text, open_mod}.</li><li>Response: _OK, I'm on the case…{nl} So I've studied the numbers and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _give me all the bills I need to pay in advance of Tues_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.HasDateCriteria to be true, but found False.</li><li>Expected to find entity 'datetime' but found only {unknown_entity, paid_query, text, all, dt, pay_type, pay_to}</li><li>Response: _I'm sorry but I didn't understand all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _is there a lot of money due in august_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.HasDateCriteria to be true, but found False.</li><li>Expected to find one of entities {bill_to, pay_from} but found only {text, open_mod}.</li><li>Response: _Just a second, I'm exploring now…{nl} So I've just looked over the finances and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _I'd like to see all the ABC Ltd invoices from the last year_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.InvoiceDateQueryStart to be <2017-12-08>, but found <2017-01-01>.</li><li>Response: _Okeydokes. Let's take a look…{nl} Righto, I've found out how things stand and you have no invoices with ABC Ltd in 2017._</li></ul> |
| _do we have any invoices overdue by more than 10 days_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.OverdueBy to be 10, but found 0.</li><li>Response: _I'm sorry but I didn't understand all of that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81._</li></ul> |
| _are any of the invoices late by more than 30 days_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.OverdueBy to be 30, but found 0.</li><li>Response: _I didn't understand all of that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _is there an invoice that's more than 30 days late_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.OverdueBy to be 30, but found 0.</li><li>Response: _Hang on - I didn't quite understand all that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _can i request the bills for 1st qtr 2016_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.PaidDateQueryStart to be <2016-01-01>, but found <0001-01-01 00:00:00&hairsp;.000>.</li><li>Response: _Okeydokes. Let's take a look…{nl} I've just reviewed your accounts and in Q1 2016 you received 12 invoices totalling £4,946.14, all of which have been paid and you raised 18 invoices totalling £6,850.26, all of which have been paid._</li></ul> |
| _do I owe anything particularly big_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected to find entity 'comparator' but found only {unknown_entity, text, paid_query, pay_to}</li><li>Response: _I'm sorry but I didn't understand all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me the most expensive one owed to me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected to find entity 'comparator' but found only {paid_query, text, pay_from, jj}</li><li>Response: _I'm sorry but I didn't understand all of that, however the amount you are owed from all clients is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _tell me who I owe the biggest amount to_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected to find entity 'comparator' but found only {who, paid_query, pay_to, text}</li><li>Response: _Right, let's see just now…{nl} I've ascertained the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _tell me who owes me the most_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected {comparator} to have single value 'biggest' but found 'most'</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _to whom do I owe the most money_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected to find entity 'comparator' but found only {unknown_entity, money, text, paid_query, pay_to}</li><li>Response: _I didn't understand all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _what is the most anyone owes me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected {comparator} entity with value 'biggest' but found only 'most'</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _what is the most substantial one owed to me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected {comparator} entity with value 'biggest' but found only 'big'</li><li>Response: _I'm afraid I didn't quite get all of that, however the amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _what's the biggest sum I'm owed_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Expected to find entity 'comparator' but found only {paid_query, pay_from, text, query_total}</li><li>Response: _Right, let me just look…{nl} I've looked into this and the amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _who billed me for the most this month_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Biggest, but found None.</li><li>Response: _Regrettably, I didn't recognise everything you said, however in December 2018 you received seven invoices totalling £2,847.15, none of which have been paid._</li></ul> |
| _which invoice has been overdue the longest_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Oldest, but found None.</li><li>Expected to find entity 'comparator' but found only {dt, pay_type, text, open_mod, overdue, unknown_entity}</li><li>Response: _I didn't understand all of that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _which invoice is most overdue_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Oldest, but found None.</li><li>Expected {comparator} entity with value 'oldest' but found only 'most'</li><li>Response: _I'm sorry but I didn't understand all of that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _which invoice is the most overdue_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Oldest, but found None.</li><li>Expected {comparator} entity with value 'oldest' but found only 'most'</li><li>Response: _Hang on - I didn't quite understand all that, however the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _show me the least valuable invoice owed to me_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Smallest, but found None.</li><li>Expected to find entity 'comparator' but found only {paid_query, text, dt, jjs, pay_type, pay_from}</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you are owed from all clients is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _what is the least expensive invoice I owe_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryRecordComparator to be Smallest, but found None.</li><li>Expected to find entity 'comparator' but found only {paid_query, pay_to, text, dt, jj, pay_type}</li><li>Response: _I'm sorry but I didn't understand all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _who's bills were the most expensive this month_ | <ul><li>Expected ctx&hairsp;.SingleContext&hairsp;.QueryType to be PaymentQuery, but found InvoiceQuery.</li><li>Expected to find entity 'comparator' but found only {unknown_entity, text, date_mod, datetime, who, pay_type}</li><li>Response: _I'm afraid I didn't quite get all of that, however in December 2018 you received seven invoices totalling £2,847.15, none of which have been paid and you raised seven invoices totalling £9,262.62, none of which have been paid._</li></ul> |
| _how many of our invoices are supplier invoices_ | <ul><li>Expected invoice state to be one of 'ExplicitAll', 'ImplicitAll' but found 'ImplicitUnpaid'.</li><li>Response: _OK, let me take a look…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me all of the abc invoices_ | <ul><li>Expected invoice state to be one of 'ExplicitAll', 'ImplicitAll' but found 'ImplicitUnpaid'.</li><li>Response: _Just a second, I'm exploring now…{nl} So I've just looked over the finances and the amount you owe to ABC Ltd is one outstanding bill which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _show me the complete set of abc invoices_ | <ul><li>Expected invoice state to be one of 'ExplicitAll', 'ImplicitAll' but found 'ImplicitUnpaid'.</li><li>Response: _Right, let me just look…{nl} All right, I've taken stock and the amount you owe to ABC Ltd is one open bill which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _show me the entire set of abc invoices_ | <ul><li>Expected invoice state to be one of 'ExplicitAll', 'ImplicitAll' but found 'ImplicitUnpaid'.</li><li>Response: _Right, let me just look…{nl} I've just queried for you and the amount you owe to ABC Ltd is one open bill which is for the amount of £22.80 and was due on the 2nd December 2018._</li></ul> |
| _all in all what do alpha owe me_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ExplicitAll'.</li><li>Response: _Let's check that out for you…{nl} I've checked that out and you have no incoming invoices payable from Alpha Ltd._</li></ul> |
| _show me all the bills I can pay_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ExplicitAll'.</li><li>Expected to find entity 'open_mod' but found only {all, dt, pay_type, text, pay_to}</li><li>Response: _Your wish is my command…{nl} I've investigated the current situation and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _are all my invoices settled for oct_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ImplicitAll'.</li><li>Response: _Let me check that out…{nl} I've performed a quick investigation and in October 2018 you received eight invoices totalling £3,761.48, of which seven have been paid totalling £2,960.64 and you raised nine invoices totalling £11,990.10, of which eight have been paid totalling £11,666.83._</li></ul> |
| _are my bills settled for december_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ImplicitAll'.</li><li>Response: _Just a second, I'm exploring now…{nl} I can confirm in December 2018 you received seven invoices totalling £2,847.15, none of which have been paid and you raised seven invoices totalling £9,262.62, none of which have been paid._</li></ul> |
| _are my invoices settled for this month_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ImplicitAll'.</li><li>Response: _I'll find out right away…{nl} I've scanned the digits and in December 2018 you received seven invoices totalling £2,847.15, none of which have been paid and you raised seven invoices totalling £9,262.62, none of which have been paid._</li></ul> |
| _do we have any bills to pay this week_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ImplicitAll'.</li><li>Response: _Right, let me just look…{nl} So I've just looked over the finances and you have no outgoing invoices this week._</li></ul> |
| _which bills should I give priority to this week_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'ImplicitAll'.</li><li>Response: _Happy to help…{nl} I can confirm you have no invoices this week._</li></ul> |
| _are there any invoices that demand immediate attention_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'Overdue'.</li><li>Expected to find entity 'open_mod' but found only {datetime_now, text, overdue, dt, pay_type}</li><li>Response: _Just a second, I'm exploring now…{nl} Well, I've checked the most up-to-date records and the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all customers is four invoices totalling £1,478.27._</li></ul> |
| _are there any invoices that need immediate attention_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'ImplicitUnpaid' but found 'Overdue'.</li><li>Expected to find entity 'open_mod' but found only {datetime_now, text, overdue, dt, pay_type}</li><li>Response: _OK, let's see…{nl} I've just checked the up-to-the-minute situation and the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all customers is four invoices totalling £1,478.27._</li></ul> |
| _show me the invoices that didn't get paid in July_ | <ul><li>Expected invoice state to be one of 'Unpaid', 'Overdue' but found 'ImplicitAll'.</li><li>Expected to find entity 'open_mod' but found only {datetime, text, dt, pay_type}</li><li>Response: _I'll look right now…{nl} I've checked that out and in July 2018 you received seven invoices totalling £2,453.54, all of which have been paid and you raised nine invoices totalling £10,466.87, all of which have been paid._</li></ul> |
| _can you see any supplier invoices that are past due by more than 20 days_ | <ul><li>Expected InvoiceState to be InvoiceState&hairsp;.Overdue but found InvoiceState&hairsp;.ImplicitUnpaid</li><li>Expected to find entity 'overdue' but found only {unknown_entity, dt, creditor, pay_type, text}</li><li>Response: _I'm afraid I didn't quite get all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do I need to pursue anyone for payments_ | <ul><li>Expected InvoiceState to be InvoiceState&hairsp;.Overdue but found InvoiceState&hairsp;.Unpaid</li><li>Expected to find one of entities {bill_to, pay_from} but found only {paid_query, pay_to, text}.</li><li>Response: _Okeydokes. Let's take a look…{nl} All right, I've taken stock and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _which bills have to be paid immediately_ | <ul><li>Expected InvoiceState to be InvoiceState&hairsp;.Overdue but found InvoiceState&hairsp;.Unpaid</li><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, text, paid_query}</li><li>Response: _OK, I'm on this now…{nl} I've scanned the digits and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which payable bills should I give priority to_ | <ul><li>Expected InvoiceState to be InvoiceState&hairsp;.Overdue but found InvoiceState&hairsp;.Unpaid</li><li>Expected to find entity 'datetime_now' but found only {dt, open_mod, pay_type, text}</li><li>Response: _Right, let me just look…{nl} I've gone through the most recent figures and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any bills I need to pay_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Expected to find entity 'open_mod' but found only {paid_query, text, dt, pay_type, pay_to}</li><li>Response: _I'll look right now…{nl} I've performed a quick investigation and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _Do I have any invoices due within the next 60 days_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Expected {date_mod} to have single value 'next' but found 'the'</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do I have any outstanding bills_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Response: _I'll look right now…{nl} I've evaluated the state of affairs and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do I have any outstanding bills from march_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Response: _Happy to help…{nl} Well, I've checked the most up-to-date records and the amount you owe to all suppliers in March 2019 is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do I have any unpaid invoices_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Response: _OK, let me take a look…{nl} I've just checked the up-to-the-minute situation and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do we have any invoices due straight away_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text, bill_from}</li><li>Response: _OK, I'm on the case…{nl} I've determined how things are. The amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _have I got any outstanding bills_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Response: _I'll look right now…{nl} I can confirm the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _have we got an invoice due to be paid on Wed_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Response: _I'll look right now…{nl} I've had a quick look and the amount you owe to all suppliers on Wednesday (5th December) is seven outstanding bills totalling £3,954.81, all of which are overdue._</li></ul> |
| _is there anything due that I need to deal with immediately_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Other than 'extras and confirmations' expected a single chunk but found 2</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _which invoice did we receive first_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.All but found QueryTransactionType&hairsp;.Outgoing</li><li>Expected to find entity 'comparator' but found only {dt, pay_type, text, bill_from}</li><li>Response: _Right, let me just look…{nl} I've inspected and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _are there any payments that need pursuing_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.Incoming but found QueryTransactionType&hairsp;.All</li><li>Expected to find entity 'incoming' but found only {overdue, paid_query, text, dt}</li><li>Response: _Let's have a peek…{nl} I've had a look at the latest picture and the overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all customers is four invoices totalling £1,478.27._</li></ul> |
| _show me my debtor invoice data_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.Incoming but found QueryTransactionType&hairsp;.All</li><li>Expected to find entity 'debtor' but found only {pay_type, text}</li><li>Response: _Ok, I'm examining that now…{nl} I've just queried for you and the amount you owe to all suppliers is nine open bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _who needs chasing up about payments_ | <ul><li>Expected QueryTransactionType to be QueryTransactionType&hairsp;.Incoming but found QueryTransactionType&hairsp;.All</li><li>Expected to find one of entities {bill_to, pay_from} but found only {who, overdue, paid_query, text}.</li><li>Response: _OK, I'm on the case…{nl} I've determined how things are. The overdue amount you owe to all suppliers is seven bills totalling £3,954.81. The overdue amount you are owed from all clients is four invoices totalling £1,478.27._</li></ul> |
| _chase down the bills that need paying this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Let me check that out…{nl} I've analysed the state of play and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _do any bills have to be paid this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Right, let's see just now…{nl} I've had a look at your account details and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _do any bills need paying this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _OK, I'm on the case…{nl} I've just reviewed your accounts and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _do any bills need sorting out this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Right, let's see just now…{nl} All right, I've taken stock and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _give me all the bills that need paying prior to Tues_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {pay_to, bill_from} but found only {unknown_adjv, datetime, text, paid_query, all, dt, pay_type}.</li><li>Response: _I'm afraid I didn't quite get all of that, however the amount you owe to all suppliers on Tuesday (4th December) is seven outstanding bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients on Tuesday (4th December) is four outstanding invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _give me every invoice that is owed_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Ok, I'm examining that now…{nl} I've evaluated the state of affairs and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _how many bills need paying this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Ok, I'm examining that now…{nl} I've had a look at the latest picture and the amount you owe to all suppliers this week is seven outstanding bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four outstanding invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _how much money is my landlord expecting this month_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {date_mod, datetime, text, money, open_mod, pay_any}.</li><li>Response: _I can't find that company in your records._</li></ul> |
| _is there anyone I have to pay_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {paid_query, text}.</li><li>Response: _OK, let's see…{nl} I've just reviewed your accounts and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _is there anyone who needs paying before a specific date_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {unknown_entity, text, who, paid_query}.</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _is there anyone who needs to be paid_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {who, paid_query, text}.</li><li>Response: _OK, let's see…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _send me a bill that needs paying by next Tues_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {pay_to, bill_from} but found only {date_mod, datetime, text, paid_query, dt, pay_type}.</li><li>Response: _I'm determining that now…{nl} I've scanned the digits and the amount you owe to all suppliers on Tuesday 11th December is seven outstanding bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients on Tuesday 11th December is four outstanding invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _show me an invoice it is possible to pay_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find entity 'open_mod' but found only {unknown_entity, text, paid_query, dt, pay_type}</li><li>Response: _I didn't understand all of that, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _show me my creditor invoice data_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find entity 'creditor' but found only {pay_type, text}</li><li>Response: _OK hold on, I'm checking now…{nl} I've analysed the state of play and the amount you owe to all suppliers is nine open bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _track down the bills that need paying this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Right, let's see just now…{nl} I've had a scoot around and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _what sort of bills need paying this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Let me check that out…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _when are my payments expected_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {when, open_mod, text}.</li><li>Response: _Righty ho, let me investigate immediately…{nl} So I've studied the numbers and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _when do my bills come due_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Let me check that out…{nl} Well, I've checked the most up-to-date records and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which bills have gotta be paid next week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _Let's check that out for you…{nl} I've gone through the most recent figures and the amount you owe to all suppliers next week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients next week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _which bills need sorting out this week_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Response: _OK, let's see…{nl} I've just checked the up-to-the-minute situation and the amount you owe to all suppliers this week is seven unpaid bills totalling £3,954.81, all of which are overdue. The amount you are owed from all clients this week is four unpaid invoices totalling £1,478.27, all of which are overdue._</li></ul> |
| _which one would be the most costly to pay_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {unknown_entity, text, paid_query}.</li><li>Response: _I'm afraid I didn't quite get all of that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which supplier invoices haven't been paid yet_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find entity 'creditor' but found only {open_mod, pay_type, text}</li><li>Response: _Let's have a peek…{nl} I can confirm the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _who still needs paying_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'All'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {who, paid_query, text}.</li><li>Response: _Let's check that out for you…{nl} Righto, I've found out how things stand and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _how much am I owing_ | <ul><li>Expected transaction type to be one of 'Outgoing', 'ImplicitOutgoing' but found 'Incoming'.</li><li>Expected to find one of entities {bill_from, pay_to} but found only {text, paid_query, pay_from}.</li><li>Response: _OK, I'm on this now…{nl} Well, I've checked the most up-to-date records and the amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |

**Comment by SME-Issues** [comment]
### Understood and failed but likely lower priority



|Expressions|Notes|
|---|---|
| _across all the invoices, how much am I owed_ | <ul><li>Expected to find entity 'all' but found only {text, paid_query, pay_from}</li><li>Response: _OK hold on, I'm checking now…{nl} Well, I've checked the most up-to-date records and the amount you are owed from all debtors is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any bills that need paying_ | <ul><li>Expected to find entity 'open_mod' but found only {paid_query, text, dt, pay_type}</li><li>Response: _Right, let me look into that…{nl} I've ascertained the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any bills which need paying_ | <ul><li>Expected to find entity 'open_mod' but found only {paid_query, text, dt, pay_type}</li><li>Response: _OK, let me take a look…{nl} I've performed a quick investigation and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices due at short notice_ | <ul><li>Expected to find entity 'datetime_now' but found only {text, dt, pay_type, open_mod}</li><li>Response: _Happy to help…{nl} I've analysed the state of play and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices due at this time_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _OK hold on, I'm checking now…{nl} So I've just looked over the finances and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices due now_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _Let me check that out…{nl} After checking, I can verify that the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices due right away_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _Right, let me just look…{nl} I've had a scoot around and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices due right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _I'm determining that now…{nl} I've analysed the state of play and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are there any invoices from alpha ltd that still need to be paid_ | <ul><li>Expected to find entity 'open_mod' but found only {text, bill_from, paid_query, dt, pay_type}</li><li>Response: _OK hold on, I'm checking now…{nl} I've just checked the up-to-the-minute situation and the amount you owe to Alpha Ltd is three unpaid bills totalling £2,273.50, all of which are overdue._</li></ul> |
| _are there any invoices with a deadline coming up_ | <ul><li>Expected to find entity 'open_mod' but found only {unknown_entity, dt, pay_type, text}</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are thre any invoices due momentarily_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _Right, let me look into that…{nl} I've checked that out and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _are thre any invoices due urgently_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _I'll find out right away…{nl} I've explored and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all customers is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _can you show me the invoices Violet is yet to pay_ | <ul><li>Expected to find one of entities {pay_from, bill_to} but found only {paid_query, text, dt, pay_type, pay_any}.</li><li>Response: _I can't find that company in your records._</li></ul> |
| _can you take a look and see how much I owe_ | <ul><li>Other than 'extras and confirmations' expected a single chunk but found 2</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _do any of our invoices have a pressing deadline_ | <ul><li>Expected to find entity 'open_mod' but found only {pay_type, text}</li><li>Response: _I'll find out right away…{nl} I've had a look at your account details and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _do we have any pressing invoices_ | <ul><li>Expected to find entity 'open_mod' but found only {dt, pay_type, text}</li><li>Response: _I'll find out right away…{nl} I've had a quick look and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _find all the payable bills and show them to me_ | <ul><li>Other than 'extras and confirmations' expected a single chunk but found 2</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _give me an invoice due in the next 60 days_ | <ul><li>Expected to find entity 'date_mod' but found only {datetime, text, dt, pay_type, open_mod}</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _give me an invoice due right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _Right, let me look into that…{nl} I can confirm the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _is anyone expecting money from me right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {paid_query, text, pay_to}</li><li>Response: _Right, let's see just now…{nl} I've scanned the digits and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _is there an invoice due immediately_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, open_mod, text}</li><li>Response: _Happy to help…{nl} After checking, I can verify that the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _is there an invoice from alpha ltd that hasn't been paid_ | <ul><li>Expected to find entity 'open_mod' but found only {text, bill_from, dt, pay_type}</li><li>Response: _Okeydokes. Let's take a look…{nl} I've gone through the most recent figures and the amount you owe to Alpha Ltd is three outstanding bills totalling £2,273.50, all of which are overdue._</li></ul> |
| _is there an invoice I need to deal with right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {paid_query, text, dt, pay_type, pay_to}</li><li>Response: _Let's check that out for you…{nl} I've evaluated the state of affairs and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _Is there anyone I need to pay right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {paid_query, text, pay_to}</li><li>Response: _I'm determining that now…{nl} I've had a look at the latest picture and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _Is there anyone I should be paying at the moment_ | <ul><li>Expected to find entity 'datetime_now' but found only {open_mod, pay_to, text}</li><li>Response: _OK, I'm on the case…{nl} I've performed a quick investigation and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me an invoice I'm able to pay_ | <ul><li>Expected to find entity 'open_mod' but found only {text, paid_query, dt, pay_type, pay_to}</li><li>Response: _OK hold on, I'm checking now…{nl} I've had a look at the latest picture and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me any invoice due within 60 days_ | <ul><li>Expected to find entity 'date_mod' but found only {datetime, text, dt, pay_type, open_mod}</li><li>Response: _Hang on - I didn't quite understand all that, however the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine outstanding invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _show me any invoice that I can pay_ | <ul><li>Expected to find entity 'open_mod' but found only {dt, pay_type, text, pay_to}</li><li>Response: _Let's have a peek…{nl} I've just queried for you and the amount you owe to all suppliers is nine outstanding bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me the invoice from alpha ltd which isn't paid_ | <ul><li>Expected to find entity 'open_mod' but found only {text, bill_from, dt, pay_type}</li><li>Response: _OK, I'm on the case…{nl} I've explored and the amount you owe to Alpha Ltd is three open bills totalling £2,273.50, all of which are overdue._</li></ul> |
| _show me the invoices I haven't got around to paying yet_ | <ul><li>Expected to find entity 'open_mod' but found only {paid_query, text, dt, pay_type, pay_to}</li><li>Response: _Let me check that out…{nl} I've determined how things are. The amount you owe to all suppliers is nine open bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me the invoices I'm yet to pay_ | <ul><li>Expected to find entity 'open_mod' but found only {paid_query, text, dt, pay_type, pay_to}</li><li>Response: _I'm determining that now…{nl} I've performed a quick investigation and the amount you owe to all suppliers is nine open bills totalling £4,431.64 of which seven are overdue._</li></ul> |
| _show me the invoices we're behind on_ | <ul><li>Expected to find entity 'open_mod' but found only {unknown_entity, text, dt, pay_type}</li><li>Response: _I'm afraid I didn't quite get all of that, however the amount you owe to all suppliers is nine open bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine open invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which invoices are due at short notice_ | <ul><li>Expected to find entity 'datetime_now' but found only {unknown_entity, text, dt, pay_type, open_mod}</li><li>Response: _Regrettably, I didn't recognise everything you said, however the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which invoices are due immediately_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, text, open_mod}</li><li>Response: _Okeydokes. Let's take a look…{nl} All right, I've taken stock and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _which invoices are due right now_ | <ul><li>Expected to find entity 'datetime_now' but found only {dt, pay_type, text, open_mod}</li><li>Response: _Ok, I'm examining that now…{nl} Well, I've checked the most up-to-date records and the amount you owe to all suppliers is nine unpaid bills totalling £4,431.64 of which seven are overdue. The amount you are owed from all clients is nine unpaid invoices totalling £9,891.97 of which four are overdue._</li></ul> |
| _who should be paid next_ | <ul><li>Expected to find one of entities {bill_from, pay_to} but found only {creditor, open_mod, comparator, text, who}.</li><li>Response: _OK hold on, I'm checking now…{nl} I've just queried for you and the oldest unpaid invoice you owe was due on the 2nd October 2018 to Alpha Ltd for the value of £800.84._</li></ul> |

**Comment by SME-Issues** [comment]
### Test Exception



|Expressions|Notes|
|---|---|
| _are any client invoices more than ten days past due?_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.ArgumentOutOfRangeException: Exception of type 'System&hairsp;.ArgumentOutOfRangeException' was thrown&hairsp;.Parameter name: Fix… |
| _are any payable invoices more than ten days past due_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.ArgumentOutOfRangeException: Exception of type 'System&hairsp;.ArgumentOutOfRangeException' was thrown&hairsp;.Parameter name: Fix… |
| _are any supplier invoices more than twenty days past due?_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.ArgumentOutOfRangeException: Exception of type 'System&hairsp;.ArgumentOutOfRangeException' was thrown&hairsp;.Parameter name: Fix… |
| _have we got any client invoices that are more than 10 days past due_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.ArgumentOutOfRangeException: Exception of type 'System&hairsp;.ArgumentOutOfRangeException' was thrown&hairsp;.Parameter name: Fix… |
| _show me any invoice due in the next 2 months_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.ArgumentOutOfRangeException: Exception of type 'System&hairsp;.ArgumentOutOfRangeException' was thrown&hairsp;.Parameter name: Fix… |
| _which accounts can be paid from 1 June to 1 July_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: found multiple day_of_month values   at PulseAPI&hairsp;.Handlers&hairsp;.DateHandler&hairsp;.TryGetDayOfMonth(… |
| _I'd be interested in a breakdown of my creditors_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Multiple creditor values found&hairsp;.   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.TestEntityPairs… |
| _can we any papa johns bills examine_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Multiple pay_any values in text 'can we any papa johns bills examine'   at PulseAPI&hairsp;.Pro… |
| _am I owing omicron any money_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Multiple pay_to values found&hairsp;.   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.TestEntityPairs(S… |
| _has anyone given us a payment deadline_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Multiple pay_to values found&hairsp;.   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.TestEntityPairs(S… |
| _who am I waiting on to pay me_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Multiple pay_to values found&hairsp;.   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.TestEntityPairs(S… |
| _are any payable invoices more than a week past due_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Unknown modifer values - 'than and a'   at PulseAPI&hairsp;.Handlers&hairsp;.DateHandler&hairsp;.GetEntities(Li… |
| _have we got any invoices more than a month past due_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Unknown modifer values - 'than and a'   at PulseAPI&hairsp;.Handlers&hairsp;.DateHandler&hairsp;.GetEntities(Li… |
| _does anyone have an account payable between 1 June and 1 July_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Unsupported intent: 'multi_query'   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.ProcessIntent… |
| _has anyone owed me for a long time_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Unsupported intent: 'multi_query'   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.ProcessIntent… |
| _show me whose accounts are payable between 1 June to 1 July_ | PulseAPI&hairsp;.Tests&hairsp;.Reporting&hairsp;.Helpers&hairsp;.TestFailedException: Error running test ---> System&hairsp;.InvalidOperationException: Unsupported intent: 'multi_query'   at PulseAPI&hairsp;.Processors&hairsp;.InputProcessor&hairsp;.ProcessIntent… |
